hadoop-plugin/src/main/groovy/com/linkedin/gradle/hadoopdsl/job/SparkJob.groovy

```diff
@@ -0,0 +1,182 @@
+ *     executes './lib/azkaban-spark-job.jar' // Required
+ *     appParams params // application parameters
+ *     enableFlags flags
+ *
+ *     set Properties: [
+ *       'master' : 'yarn-cluster',
+ *       'executor-memory' : '1g'
+ *     ]
+ *
+ *     set confs: [
+ *       'key1' : 'val1',
+ *       'key2' : 'val2'
+ *     ]
+ *
+ *   }
+ * </pre>
+ */
+
+class SparkJob extends HadoopJavaProcessJob {
+  Map<String, Object> properties;
+  Map<String, Object> confs;
+  List<String> flags;
+  List<String> appParams;
+  String appClass;
+  String executionJar;
+
+  /**
+   * Constructor for a SparkJob.
+   *
+   * @param jobName The job name
+   */
+  SparkJob(String jobName) {
+    super(jobName);
+    properties = new LinkedHashMap<String, Object>();
+    confs = new LinkedHashMap<String, Object>();
+    flags = new LinkedList<String>();
+    appParams = new LinkedList<String>()
+    setJobProperty("type", "spark");
+  }
+
+  /**
+   * Clones the job.
+   *
+   * @return The cloned job
+   */
+  @Override
+  SparkJob clone() {
+    return clone(new SparkJob(name));
+  }
+
+  /**
+   * Helper method to set the properties on a cloned job.
+   *
+   * @param cloneJob The job being cloned
+   * @return The cloned job
+   */
+  SparkJob clone(SparkJob cloneJob) {
+    cloneJob.properties.putAll(properties);
+    cloneJob.confs.putAll(confs);
+    cloneJob.flags.addAll(flags);
+    cloneJob.appParams.addAll(appParams);
+    cloneJob.appClass = appClass;
+    cloneJob.executionJar = executionJar;
+    return super.clone(cloneJob);
+  }
+
+  /**
+   * DSL method to specify job properties to set on the job. Specifying job properties causes lines
+   * of the form key=val to be written to the job file.
+   * <p>
+   * Additionally for JavaProcessJobs, you can specify JVM properties by using the syntax
+   * "set jvmProperties: [ ... ]", which causes a line of the form
+   * jvm.args=-Dkey1=val1 ... -DkeyN=valN to be written to the job file.
+   * <p>
+   * Additionally for HadoopJavaProcessJob subclasses, you can specify Hadoop job configuration
+   * properties by using the syntax "set confProperties: [ ... ]", which causes lines of the form
+   * hadoop-inject.key=val to be written to the job file.
+   * <p>
+   * Additionally for SparkJobs, you can specify Spark configuration by using the syntax
+   * "set confs: [ ... ]". For each parameter you set, a line of the form conf.key=val will
+   * be written to the job file.
+   *
+   * @param args Args whose key 'properties' has a map value specifying the job properties to set;
+   *   or a key 'jvmProperties' with a map value that specifies the JVM properties to set;
+   *   or a key 'confProperties' with a map value that specifies the Hadoop job configuration properties to set;
+   *   or a key 'confs' with a map value that specifies the Spark configuration to set
+   */
+  @Override
+  void set(Map args) {
+    super.set(args);
+    if (args.containsKey("confs")) {
```

Comment:
At the Hadoop DSL level, I would be in favor of naming this something just a little more descriptive - maybe "set sparkConfs:"? Is that better than "set confs:"???